marconi-chain-index/src/Marconi/ChainIndex/Experimental/Indexers/Utxo.hs

```diff
@@ -0,0 +1,699 @@
+-- instance (MonadIO m, MonadError (Core.QueryError QueryUtxoByAddress) m)
+instance (MonadIO m)
+    => Core.Queryable m UtxoEvent QueryUtxoByAddress Core.SQLiteIndexer where
+  -- query
+  --   :: (MonadError (Core.QueryError QueryUtxoByAddress) m, Ord (Core.Point UtxoEvent))
+  --   => Core.Point UtxoEvent
+  --   -> QueryUtxoByAddress
+  --   -> Core.SQLiteIndexer UtxoEvent
+  --   -> m (Core.Result QueryUtxoByAddress)
+  query cp q (Core.SQLiteIndexer conn _ _ _) =
+    let
+
+      action :: SQL.Connection -> IO (Core.Result QueryUtxoByAddress)
+      action = mkUtxoAddressQueryAction cp q
+
+    in liftIO $ action conn
+
+-- | Here we define our QueryAction functions
+mkUtxoAddressQueryAction
+  :: MonadIO m
+  => Core.Point UtxoEvent -- ^ chain point to compute/query future spent
+  -> QueryUtxoByAddress -- ^ Query
+  -> (SQL.Connection -> m (Core.Result QueryUtxoByAddress)) -- we rerturn a query action funtion
+mkUtxoAddressQueryAction C.ChainPointAtGenesis _ = \_ -> pure []
+mkUtxoAddressQueryAction (C.ChainPoint futureSpentSlotNo _) (QueryUtxoByAddress (addr, slotNo)) =
+  let
+
+    _ = futureSpentSlotNo  -- TODO this is place holder to build the `Future Spent` as required by SideChain
+    filterPairs :: ([SQL.Query], [NamedParam])
+    filterPairs = (["u.address = :address"], [":address" := addr])
+      <> maybe mempty (\sno -> (["u.slotNo <= :slotNo"] , [":slotNo" := sno])) slotNo
+      -- TODO build the future spent in the above filer
+
+    mkUtxoAddressQueryAction'
+      :: MonadIO m
+      => [SQL.Query] -- ^ the filter part of the query
+      -> [NamedParam]
+      -> (SQL.Connection -> m (Core.Result QueryUtxoByAddress) )
+    mkUtxoAddressQueryAction' filters params=
+      let builtQuery =
+                [r|SELECT
+                      u.address,
+                      u.txId,
+                      u.txIx,
+                      u.datum,
+                      u.datumHash,
+                      u.value,
+                      u.inlineScript,
+                      u.inlineScriptHash,
+                      u.slotNo,
+                      u.blockHash
+                  FROM
+                      unspent_transactions u
+                  LEFT JOIN spent s ON u.txId = s.txId
+                  AND u.txIx = s.txIx
+                  WHERE
+                      s.txId IS NULL
+                      AND s.txIx IS NULL
+                  AND |] <> SQL.Query (Text.intercalate " AND " $ SQL.fromQuery <$> filters) <>
+                [r| ORDER BY
+                    u.slotNo ASC |]
+
+      in \conn -> liftIO $ SQL.queryNamed conn builtQuery params
+
+  in uncurry mkUtxoAddressQueryAction' filterPairs
+
+instance MonadIO m => Core.Rewindable m UtxoEvent Core.SQLiteIndexer where
+
+    rewind C.ChainPointAtGenesis ix = do
+      let c = ix ^. Core.handle
+      liftIO $ SQL.execute_ c "DROP TABLE IF EXISTS unspent_transactions"
+      liftIO $ SQL.execute_ c "DROP TABLE IF EXISTS spent"
```

Comment:
Do we really want to drop the table here? It will break future indexing operations, no?

Reply:
It will. I'll change to delete. However, I am not certain if we want to apply the same database indexing as your ticket , PLT-5658[https://input-output.atlassian.net/browse/PLT-5658](url) here. My guess is probably not, not at least at the beginning